app/assets/javascripts/mobile/mobile_file_uploader.js

```diff
@@ -4,78 +4,91 @@
+      onComplete: function(_id, fileName, responseJSON) {
+        $("#fileInfo-publisher").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
         var id = responseJSON.data.photo.id,
             url = responseJSON.data.photo.unprocessed_image.url,
-            currentPlaceholder = $('li.loading').first();
+            currentPlaceholder = $("li.loading").first();
 
-        $('#publisher_textarea_wrapper').addClass("with_attachments");
-        $('#new_status_message').append("<input type='hidden' value='" + id + "' name='photos[]' />");
+        $("#publisher_textarea_wrapper").addClass("with_attachments");
+        $("#new_status_message").append("<input type='hidden' value='" + id + "' name='photos[]' />");
 
         // replace image placeholders
-        var img = currentPlaceholder.find('img');
-        img.attr('src', url);
-        img.attr('data-id', id);
-        currentPlaceholder.removeClass('loading');
+        var img = currentPlaceholder.find("img");
+        img.attr("src", url);
+        img.attr("data-id", id);
+        currentPlaceholder.removeClass("loading");
         currentPlaceholder.append("<div class='x'>X</div>" +
-            "<div class='circle'></div>");
-        ////
+          "<div class='circle'></div>");
 
-        var publisher = $('#publisher');
+        var publisher = $("#publisher");
 
-        publisher.find("input[type='submit']").removeAttr('disabled');
+        publisher.find("input[type='submit']").removeAttr("disabled");
 
-        $('.x').bind('click', function(){
-          var photo = $(this).closest('.publisher_photo');
+        $(".x").bind("click", function() {
+          var photo = $(this).closest(".publisher_photo");
           photo.addClass("dim");
-          $.ajax({url: "/photos/" + photo.children('img').attr('data-id'),
-                  dataType: 'json',
-                  type: 'DELETE',
-                  success: function() {
-                            photo.fadeOut(400, function(){
-                              photo.remove();
-                              if ( $('.publisher_photo').length === 0){
-                                $('#publisher_textarea_wrapper').removeClass("with_attachments");
-                              }
-                            });
-                          }
-                  });
+          $.ajax({
+            url: "/photos/" + photo.children("img").attr("data-id"),
+            dataType: "json",
+            type: "DELETE",
+            success: function() {
+              photo.fadeOut(400, function() {
+                photo.remove();
+                if ($(".publisher_photo").length === 0) {
+                  $("#publisher_textarea_wrapper").removeClass("with_attachments");
+                }
+              });
+            }
+          });
         });
-       },
-
-       onAllComplete: function(){}
-   });
+      },
+      onError: function(id, name, errorReason) {
+        alert(errorReason);
+      }
```

Comment:
I still don't like the alerts. ;) Again, `onComplete` will also be called on errors, so like above, either do the error handling there or do it in `onError` and make sure that you don't do stuff in `onComplete` that one would only do after a successful upload.

(optional, at least remove the alert)